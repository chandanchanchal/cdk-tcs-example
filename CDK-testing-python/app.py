#!/usr/bin/env python3
import os

from aws_cdk import core as cdk


app = cdk.App()

# Stacks are intentionally not created here -- this application isn't meant to
# be deployed.

app.synth()
