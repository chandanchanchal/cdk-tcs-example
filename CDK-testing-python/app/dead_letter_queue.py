from aws_cdk import aws_cloudwatch as cloudwatch
from aws_cdk import aws_sqs as sqs
from aws_cdk import core as cdk


class DeadLetterQueue(sqs.Queue):
    def __init__(self, scope: cdk.Construct, id: str):
        super().__init__(scope, id)

        self.messages_in_queue_alarm = cloudwatch.Alarm(
            self,
            "Alarm",
            alarm_description="There are messages in the Dead Letter Queue.",
            evaluation_periods=1,
            threshold=1,
            metric=self.metric_approximate_number_of_messages_visible(),
        )
