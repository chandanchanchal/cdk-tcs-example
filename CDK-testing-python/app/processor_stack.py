from typing import List

from aws_cdk import aws_lambda as lambda_
from aws_cdk import aws_sns as sns
from aws_cdk import aws_sns_subscriptions as sns_subscriptions
from aws_cdk import aws_stepfunctions as sfn
from aws_cdk import core as cdk


class ProcessorStack(cdk.Stack):
    def __init__(
        self,
        scope: cdk.Construct,
        construct_id: str,
        *,
        topics: List[sns.Topic],
        **kwargs
    ) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # In the future this state machine will do some work...
        state_machine = sfn.StateMachine(
            self, "StateMachine", definition=sfn.Pass(self, "StartState")
        )

        # This Lambda function starts the state machine.
        func = lambda_.Function(
            self,
            "LambdaFunction",
            runtime=lambda_.Runtime.NODEJS_14_X,
            handler="handler",
            code=lambda_.Code.from_asset("./start-state-machine"),
            environment={
                "STATE_MACHINE_ARN": state_machine.state_machine_arn,
            },
        )
        state_machine.grant_start_execution(func)

        subscription = sns_subscriptions.LambdaSubscription(func)
        for topic in topics:
            topic.add_subscription(subscription)
