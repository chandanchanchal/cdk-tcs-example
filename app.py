#!/usr/bin/env python3
import os

import aws_cdk as cdk

from cdk_vpc_ec2.cdk_vpc_ec2_stack import CdkVpcEc2Stack
from cdk_vpc_ec2.cdk_rds_stack import CdkRdsStack
from cdk_vpc_ec2.cdk_ec2_stack import CdkEc2Stack

app = cdk.App()

vpc_stack = CdkVpcEc2Stack(app, "cdk-vpc")


ec2_stack = CdkEc2Stack(app, "cdk-ec2",vpc=vpc_stack.vpc)
rds_stack = CdkRdsStack(app, "cdk-rds",vpc=vpc_stack.vpc,
                        asg_security_groups=ec2_stack.asg.connections.security_groups)
app.synth()
