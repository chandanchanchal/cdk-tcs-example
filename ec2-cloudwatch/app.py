#!/usr/bin/env python3
import os

import aws_cdk as cdk

from ec2_cloudwatch.ec2_cloudwatch_stack import Ec2CloudwatchStack


app = cdk.App()
Ec2CloudwatchStack(app, "Ec2CloudwatchStack")

app.synth()
