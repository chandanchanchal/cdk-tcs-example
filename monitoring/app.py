#!/usr/bin/env python3
import os

import aws_cdk as cdk

from monitoring.monitoring_stack import MonitoringStack
from monitoring.custom_cloudwatch_live_dashboard import CustomCloudwatchLiveDashboardStack
from monitoring.custom_cloudwatch_metrics import CustomMetricsStack
from monitoring.custom_ec2_with_alarms import CustomEc2WithAlarmsStack



app = cdk.App()
CustomEc2WithAlarmsStack(app, "CustomEc2WithAlarmsStack")
CustomMetricsStack(app, "CustomMetricsStack")
CustomCloudwatchLiveDashboardStack(app, "CustomCloudwatchLiveDashboardStack")


app.synth()
