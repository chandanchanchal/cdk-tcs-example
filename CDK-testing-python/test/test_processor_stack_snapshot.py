from aws_cdk import aws_sns as sns
from aws_cdk import core as cdk
from aws_cdk.assertions import Template

from app.processor_stack import ProcessorStack


# The snapshot parameter is injected by Pytest -- it's a fixture provided by
# syrupy, the snapshot testing library we're using:
# https://docs.pytest.org/en/stable/fixture.html
def test_matches_snapshot(snapshot):
    # Set up the app and resources in the other stack.
    app = cdk.App()
    topics_stack = cdk.Stack(app, "TopicsStack")
    topics = [sns.Topic(topics_stack, "Topic1")]

    # Create the ProcessorStack.
    processor_stack = ProcessorStack(
        app, "ProcessorStack", topics=topics  # Cross-stack reference
    )

    # Prepare the stack for assertions.
    template = Template.from_stack(processor_stack)

    assert template.to_json() == snapshot
