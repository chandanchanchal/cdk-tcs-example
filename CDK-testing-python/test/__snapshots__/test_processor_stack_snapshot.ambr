# name: test_matches_snapshot
  <class 'dict'> {
    'Parameters': <class 'dict'> {
      'AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586ArtifactHash97E3F10B': <class 'dict'> {
        'Description': 'Artifact hash for asset "6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586"',
        'Type': 'String',
      },
      'AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3BucketE40C5F0B': <class 'dict'> {
        'Description': 'S3 bucket for asset "6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586"',
        'Type': 'String',
      },
      'AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3VersionKey189FBF34': <class 'dict'> {
        'Description': 'S3 key for asset version "6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586"',
        'Type': 'String',
      },
    },
    'Resources': <class 'dict'> {
      'LambdaFunctionAllowInvokeTopicsStackTopic1A54934186CE533EE': <class 'dict'> {
        'Properties': <class 'dict'> {
          'Action': 'lambda:InvokeFunction',
          'FunctionName': <class 'dict'> {
            'Fn::GetAtt': <class 'list'> [
              'LambdaFunctionBF21E41F',
              'Arn',
            ],
          },
          'Principal': 'sns.amazonaws.com',
          'SourceArn': <class 'dict'> {
            'Fn::ImportValue': 'TopicsStack:ExportsOutputRefTopic198E71B3E74722020',
          },
        },
        'Type': 'AWS::Lambda::Permission',
      },
      'LambdaFunctionBF21E41F': <class 'dict'> {
        'DependsOn': <class 'list'> [
          'LambdaFunctionServiceRoleDefaultPolicy32EEEE35',
          'LambdaFunctionServiceRoleC555A460',
        ],
        'Properties': <class 'dict'> {
          'Code': <class 'dict'> {
            'S3Bucket': <class 'dict'> {
              'Ref': 'AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3BucketE40C5F0B',
            },
            'S3Key': <class 'dict'> {
              'Fn::Join': <class 'list'> [
                '',
                <class 'list'> [
                  <class 'dict'> {
                    'Fn::Select': <class 'list'> [
                      0,
                      <class 'dict'> {
                        'Fn::Split': <class 'list'> [
                          '||',
                          <class 'dict'> {
                            'Ref': 'AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3VersionKey189FBF34',
                          },
                        ],
                      },
                    ],
                  },
                  <class 'dict'> {
                    'Fn::Select': <class 'list'> [
                      1,
                      <class 'dict'> {
                        'Fn::Split': <class 'list'> [
                          '||',
                          <class 'dict'> {
                            'Ref': 'AssetParameters6ce9cba0c9785156f59d00159e27431182962b409d385ab1d5f199541a970586S3VersionKey189FBF34',
                          },
                        ],
                      },
                    ],
                  },
                ],
              ],
            },
          },
          'Environment': <class 'dict'> {
            'Variables': <class 'dict'> {
              'STATE_MACHINE_ARN': <class 'dict'> {
                'Ref': 'StateMachine2E01A3A5',
              },
            },
          },
          'Handler': 'handler',
          'Role': <class 'dict'> {
            'Fn::GetAtt': <class 'list'> [
              'LambdaFunctionServiceRoleC555A460',
              'Arn',
            ],
          },
          'Runtime': 'nodejs14.x',
        },
        'Type': 'AWS::Lambda::Function',
      },
      'LambdaFunctionServiceRoleC555A460': <class 'dict'> {
        'Properties': <class 'dict'> {
          'AssumeRolePolicyDocument': <class 'dict'> {
            'Statement': <class 'list'> [
              <class 'dict'> {
                'Action': 'sts:AssumeRole',
                'Effect': 'Allow',
                'Principal': <class 'dict'> {
                  'Service': 'lambda.amazonaws.com',
                },
              },
            ],
            'Version': '2012-10-17',
          },
          'ManagedPolicyArns': <class 'list'> [
            <class 'dict'> {
              'Fn::Join': <class 'list'> [
                '',
                <class 'list'> [
                  'arn:',
                  <class 'dict'> {
                    'Ref': 'AWS::Partition',
                  },
                  ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
                ],
              ],
            },
          ],
        },
        'Type': 'AWS::IAM::Role',
      },
      'LambdaFunctionServiceRoleDefaultPolicy32EEEE35': <class 'dict'> {
        'Properties': <class 'dict'> {
          'PolicyDocument': <class 'dict'> {
            'Statement': <class 'list'> [
              <class 'dict'> {
                'Action': 'states:StartExecution',
                'Effect': 'Allow',
                'Resource': <class 'dict'> {
                  'Ref': 'StateMachine2E01A3A5',
                },
              },
            ],
            'Version': '2012-10-17',
          },
          'PolicyName': 'LambdaFunctionServiceRoleDefaultPolicy32EEEE35',
          'Roles': <class 'list'> [
            <class 'dict'> {
              'Ref': 'LambdaFunctionServiceRoleC555A460',
            },
          ],
        },
        'Type': 'AWS::IAM::Policy',
      },
      'LambdaFunctionTopic1461E184C': <class 'dict'> {
        'Properties': <class 'dict'> {
          'Endpoint': <class 'dict'> {
            'Fn::GetAtt': <class 'list'> [
              'LambdaFunctionBF21E41F',
              'Arn',
            ],
          },
          'Protocol': 'lambda',
          'TopicArn': <class 'dict'> {
            'Fn::ImportValue': 'TopicsStack:ExportsOutputRefTopic198E71B3E74722020',
          },
        },
        'Type': 'AWS::SNS::Subscription',
      },
      'StateMachine2E01A3A5': <class 'dict'> {
        'DependsOn': <class 'list'> [
          'StateMachineRoleB840431D',
        ],
        'Properties': <class 'dict'> {
          'DefinitionString': '{"StartAt":"StartState","States":{"StartState":{"Type":"Pass","End":true}}}',
          'RoleArn': <class 'dict'> {
            'Fn::GetAtt': <class 'list'> [
              'StateMachineRoleB840431D',
              'Arn',
            ],
          },
        },
        'Type': 'AWS::StepFunctions::StateMachine',
      },
      'StateMachineRoleB840431D': <class 'dict'> {
        'Properties': <class 'dict'> {
          'AssumeRolePolicyDocument': <class 'dict'> {
            'Statement': <class 'list'> [
              <class 'dict'> {
                'Action': 'sts:AssumeRole',
                'Effect': 'Allow',
                'Principal': <class 'dict'> {
                  'Service': <class 'dict'> {
                    'Fn::Join': <class 'list'> [
                      '',
                      <class 'list'> [
                        'states.',
                        <class 'dict'> {
                          'Ref': 'AWS::Region',
                        },
                        '.amazonaws.com',
                      ],
                    ],
                  },
                },
              },
            ],
            'Version': '2012-10-17',
          },
        },
        'Type': 'AWS::IAM::Role',
      },
    },
  }
---
